pkg/sharing/placeholder_secret_test.go

```diff
@@ -0,0 +1,270 @@
+
+		reconcileObject(t, secretExportReconciler, &secretExport1)
+		reconcileObject(t, secretExportReconciler, secretExport2)
+		reconcileObject(t, secretReconciler, &placeholderSecret)
+
+		reload(t, &placeholderSecret, k8sClient)
+
+		var observedStatus map[string]interface{}
+		require.NoError(t, json.Unmarshal([]byte(placeholderSecret.ObjectMeta.Annotations["secretgen.carvel.dev/status"]), &observedStatus))
+		expectedStatus := map[string]interface{}{"conditions": []interface{}{map[string]interface{}{"status": "True", "type": "ReconcileSucceeded"}}, "secretNames": []interface{}{"test-source/test-secret", "test-source/test-secret-2"}}
+		assert.Equal(t, expectedStatus, observedStatus)
+		expectedData := []byte(`{"auths":{"server":{"username":"correctUser","password":"correctPassword","auth":"correctAuth"},"server2":{"username":"correctUser2","password":"correctPassword2","auth":"correctAuth2"}}}`)
+		assert.Equal(t, expectedData, placeholderSecret.Data[corev1.DockerConfigJsonKey])
+	})
+}
+
+func buildSourceSecret() corev1.Secret {
+	return corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-secret",
+			Namespace: "test-source",
+		},
+		Data: map[string][]byte{
+			corev1.DockerConfigJsonKey: []byte(`{"auths":{"server":{"username":"correctUser","password":"correctPassword","auth":"correctAuth"}}}`),
+		},
+		Type: "kubernetes.io/dockerconfigjson",
+	}
+
+}
+
+func sourceAndPlaceholder() (sourceSecret corev1.Secret, placeholderSecret corev1.Secret) {
+	sourceSecret = buildSourceSecret()
+	placeholderSecret = corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        "placeholder-secret",
+			Namespace:   "test-target",
+			Annotations: map[string]string{"secretgen.carvel.dev/image-pull-secret": ""},
+		},
+		Data: map[string][]byte{
+			corev1.DockerConfigJsonKey: []byte(`{"auths":{}}`),
+		},
+		Type: "kubernetes.io/dockerconfigjson",
+	}
+	return
+}
+
+func secretExportFor(sourceSecret corev1.Secret, toNamespace string) sg2v1alpha1.SecretExport {
+	return sg2v1alpha1.SecretExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      sourceSecret.Name,
+			Namespace: sourceSecret.Namespace,
+		},
+		Spec: sg2v1alpha1.SecretExportSpec{
+			ToNamespaces: []string{toNamespace},
+		},
+	}
+}
+
+func placeholderReconcilers(objects ...runtime.Object) (secretExportReconciler *sharing.SecretExportReconciler, secretReconciler *sharing.SecretReconciler, k8sClient client.Client) {
+	k8sClient = fakeClient.NewFakeClient(objects...)
+	secretExports := sharing.NewSecretExportsWarmedUp(sharing.NewSecretExports(testLogr))
+	secretExportReconciler = sharing.NewSecretExportReconciler(k8sClient, secretExports, testLogr)
+	secretExports.WarmUpFunc = secretExportReconciler.WarmUp
+	secretReconciler = sharing.NewSecretReconciler(k8sClient, secretExports, testLogr)
+	return
+}
+
+// reload asks the Kubernetes runtime client to re-populate our object
```

Comment:
this and functions below should probably be split off into their own file.

Reply:
`reload` specifically is only used in this test, but i did split out a `helpers_for_test.go` for common fns.     I'm open to other file names;  I was getting a compiler error complaining about multiple namespaces in the same folder if it didn't end in `_test.go`